src/Services/PaymentsService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace EscolaLms\Payments\Services;
+
+use EscolaLms\Payments\Models\TransactionRegistration;
+use EscolaLms\Payments\Services\Contracts\PaymentsServiceContract;
+use Illuminate\Support\Facades\Auth;
+
+
+class PaymentsService implements PaymentsServiceContract
+{
+    public function registerTransaction(int $amount, string $currency, string $description): TransactionRegistration
+    {
+        $registration = TransactionRegistration::factory()->newModel([
+            'amount' => $amount,
+            'currency' => $currency,
+            'description' => $description,
+            'buyer_id' => Auth::user()->id,
+        ]);
+        $ok = $registration->save();
+        if (!$ok) {
```

Comment:
Is this variable needed here?
```suggestion
        if (!$registration->save()) {
```

Reply:
Not really, I'm removing it.